src/main/app/idam/oAuthHelper.ts

```diff
@@ -0,0 +1,46 @@
+import * as config from 'config'
+import * as uuid from 'uuid'
+import * as Cookies from 'cookies'
+import * as express from 'express'
+import { buildURL } from 'utils/callbackBuilder'
+import { Paths } from 'app/paths'
+import { RoutablePath } from 'common/router/routablePath'
+
+const clientId = config.get<string>('oauth.clientId')
+
+export class OAuthHelper {
```

Comment:
`getRedirectUri `, `getRedirectUriForPin` and `getRedirectUriForUplift` differ only by the function which is invoked to build redirect URI and the IDAM page URI. I'd extract the common denominator.

Reply:
also differ on what goes in the state cookie

Reply:
I'd at least extract that redirect string.

Reply:
will do